File: ang2springBootPrimeNgStarter/ng2boot-master/frontend/src/main/frontend/src/app/showcase/service/vehicle.service.ts
import { Injectable } from '@angular/core';
import { Http, Response , Headers, RequestOptions} from '@angular/http';
import 'rxjs/add/operator/toPromise';
import 'rxjs/Rx';

@Injectable()
export class VehicleService {
    
    constructor(private http: Http) {}
  
    getVehicle(typeName: String, manufacturerName: String, modelName: String) {
      return this.http.get('api/vehicles/' + typeName + '/' + manufacturerName + '/' + modelName)
        .do( res => console.log('HTTP response:', res))
        .map(res => res.json());
    }

    getVehicleManufacturersByType(typeName: String) {
      return this.http.get('api/vehicle-manufacturers/' + typeName)
      .do( res => console.log('HTTP response:', res))
      .map(res => res.json());
    }

    getVehicleModelsByManufacturer(typeName: String, manufacturerName: String) {
      return this.http.get('api/vehicle-models/' + typeName + '/' + manufacturerName)
      .do( res => console.log('HTTP response:', res))
      .map(res => res.json());
    }

}

File: ang2springBootPrimeNgStarter/ng2boot-master/backend/src/main/java/com/web/project/service/VehicleService.java
package com.web.project.service;

import java.util.List;

import com.web.project.model.Vehicle;
import com.web.project.model.VehicleManufacturer;
import com.web.project.model.VehicleModel;

public interface VehicleService {
	
	public List<Vehicle> findAll();
	
	public List<Vehicle> getVehiclesByModelAndManufacturer(String manufacturerName, String modelName, String typeName);
	
	public List<VehicleModel> getVehicleModelsByManufacturer(String manufacturerName, String typeName);
	
	public List<VehicleManufacturer> getVehicleManufacturersByType(String typeName);

}
